_projects/rust-sdk-python-bindings

```diff
@@ -0,0 +1,41 @@
+---
+name: Matrix-Rust-SDK-Python
+desc: Building Python/nio bindings for using the matrix rust sdk
+requirements:
+# Student requirements:
+ - Worked with Rust FFI
+ - Familiar with building Python Bindings
+difficulty: low
+issues:
+# Related issues (if any)  to this project.
+mentors:
+# First person in contact; mentors may change before project starts.
+ - Poljar
+initiatives:
+ - GSoC
+tags:
+# Different technologies needed
+ - python
+ - Rust
+ - FFI
+ - Nio
+---
+This is an awesome project idea.
+
+#### Description
+
+[matrix-nio](https://github.com/poljar/matrix-nio) the python library to connect to matrix, lacks some important features - most notably cross-signing-support - that the new [Matrix Rust SDK](https://github.com/matrix-org/matrix-rust-sdk) supports. This projects want to create python bindings for the Rust SDK and integrate it in the nio library.
```

Comment:
Let's mention that a state store is also lacking from nio and the rust-sdk would take care of that.

>  [matrix-nio](https://github.com/poljar/matrix-nio), a Python Matrix client library, lacks some important features, up-to-date cryptography, the ability to store state, support for reactions, etc. All of those missing features are supported by the new [Matrix Rust SDK](https://github.com/matrix-org/matrix-rust-sdk).

> This project aims to create Python bindings for the Rust SDK and integrate them in the matrix-nio library. matrix-nio already contains multiple `Client` classes for different use-cases, the `AsyncClient` class API is similar to the Rust SDK API. The project should introduce a new `Client` class to `matrix-nio` which would replace the `AsyncClient` class.

Reply:
```suggestion
[matrix-nio](https://github.com/poljar/matrix-nio), a Python Matrix client library, lacks some important features, up-to-date cryptography, the ability to store state, support for reactions, etc. All of those missing features are supported by the new [Matrix Rust SDK](https://github.com/matrix-org/matrix-rust-sdk).

This project aims to create Python bindings for the Rust SDK and integrate them in the matrix-nio library. matrix-nio already contains multiple `Client` classes for different use-cases, the `AsyncClient` class API is similar to the Rust SDK API. The project should introduce a new `Client` class to `matrix-nio` which would replace the `AsyncClient` class.
```